pkg/config/decode/decode.go

```diff
@@ -0,0 +1,38 @@
+package decode
+
+import (
+	configv1 "github.com/openshift/api/legacyconfig/v1"
+	k8sjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func init() {
+	configv1.InstallLegacy(scheme.Scheme)
+}
+
+// MasterConfig unmarshals OCP3 Master
+// There is no known differences between OCP3 minor versions of the master config
+func MasterConfig(content []byte) (*configv1.MasterConfig, error) {
+	var masterConfig = new(configv1.MasterConfig)
+
+	serializer := k8sjson.NewYAMLSerializer(k8sjson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	_, _, err := serializer.Decode(content, nil, masterConfig)
```

Comment:
Interesting seeing the use of `new()`. I've always done

```golang
masterConfig := configv1.MasterConfig{}
...
serializer.Decode(content, nil, &masterConfig)
```
That is clearly a style preference, no need to change, just interesting to see your style.

Reply:
In this case I don't really see it as a different style approach.
Because when a pointer is needed `new` provides one to use along.
No need to remember this is not a pointer so one must use the address.